src/generators/n8n.ts

```diff
@@ -0,0 +1,172 @@
+  filename?: string;
+};
+
+type n8nOutput = {
+  authentication?: string;
+  genericAuthType?: string;
+  requestMethod: string;
+  url: string;
+  allowUnauthorizedCerts?: boolean;
+  options: { [key: string]: string };
+  jsonParameters: boolean;
+  sendBinaryData?: boolean;
+  binaryPropertyName?: string;
+  bodyParametersJson?: string;
+  headerParametersJson?: string;
+  queryParametersJson?: string;
+  bodyParametersUi?: { [key: string]: Array<{ [key: string]: string | null }> };
+  headerParametersUi?: {
+    [key: string]: Array<{ [key: string]: string | null }>;
+  };
+  queryParametersUi?: {
+    [key: string]: Array<{ [key: string]: string | null }>;
+  };
+};
+
+export const _toN8n = (request: Request, _warnings: Warnings = []): string => {
+  const requestJson: n8nOutput = {
+    url: (request.queryDict ? request.urlWithoutQuery : request.url).replace(
+      /\/$/,
+      ""
+    ),
+    options: {},
+    jsonParameters: true,
+    requestMethod: request.method.toUpperCase(),
+  };
+
+  console.log(JSON.stringify(request, null, 2));
+
+  if (!request.data && !request.isDataBinary && !request.multipartUploads) {
+    requestJson.jsonParameters = false;
+    if (request.headers) {
+      requestJson.headerParametersUi = {
+        parameter: request.headers.map((x) => ({ name: x[0], value: x[1] })),
+      };
+    }
+    if (request.query) {
+      requestJson.queryParametersUi = {
+        parameter: request.query.map((x) => ({ name: x[0], value: x[1] })),
+      };
+    }
+  } else {
+    if (request.isDataBinary) {
+      console.log("isDataBinary");
+      if ((request.data as string).charAt(0) === "@") {
+        requestJson.options.bodyContentType = "raw";
+        requestJson.sendBinaryData = true;
+        requestJson.binaryPropertyName = request.data;
+      } else {
+        requestJson.bodyParametersJson = request.data;
+      }
+    } else if (request.multipartUploads) {
+      console.log("multipartUploads");
+      requestJson.options.bodyContentType = "multipart-form-data";
+      requestJson.sendBinaryData = true;
+      requestJson.binaryPropertyName = request.multipartUploads
+        .filter((x: multipartContent) => x.filename)
+        .map(
+          (x: multipartContent) =>
+            `${x.name}:${x.filename || x.contentFile || x.content}`
+        )
+        .join(",");
+    } else {
+      console.log("Other");
+      const contentType = util.getContentType(request)?.toLowerCase();
+      switch (contentType) {
+        case "application/json":
+          requestJson.options.bodyContentType = "json";
+          break;
+        case "application/x-www-form-urlencoded":
+          requestJson.options.bodyContentType = "form-urlencoded";
+          break;
+        case "multipart/form-data":
+          requestJson.options.bodyContentType = "multipart-form-data";
+          break;
+      }
+      try {
+        requestJson.bodyParametersJson = JSON.stringify(
+          JSON.parse(request.data || "}{")
```

Comment:
```suggestion
          JSON.parse(request.data || "{}")
```

Reply:
This is a hack 😂. I want to force the catch to happen instead of thinking an empty or null body is valid JSON. If that happens we then try to interpret the request.data as form-data KVs. There is probably a much cleaner way to do this, hence WIP. I should probably just do it in the switch statement.